### PR TITLE
Author as reviewer

### LINKED ISSUE #34: Config option: Count author as reviewer
Add a custom option per rule to count the author as a positive review if he belongs to the teams of reviewers

### PR DESCRIPTION
**Pull request by Bullrich**
Resolves #34

Created the option to count the author of a PR as an approval.
This has to be set per review option.

In the case of the "normal" rules it doesn't change the logic.

In the case of the "and-distinct" rule we have to decide if we filter the author from the PR on evaluation time.

### CODE REVIEW

File: src/runner.ts [hunk 1]
```diff
@@ -225,13 +225,18 @@ export class ActionRunner {
...
+      });
     }
 
     // We count how many reviews are needed in total
     const requiredAmountOfReviews = rule.reviewers.map((r) => r.min_approvals).reduce((a, b) => a + b, 0);
-    // We get the list of users that approved the PR
-    const approvals = await this.prApi.listApprovedReviewsAuthors();
+    // We get the list of users that approved the PR and the author
+    // we filter the author later
+    const approvals = await this.prApi.listApprovedReviewsAuthors(true);
```
Comment (rzadp) [inline src/runner.ts:239]: I feel like this mishmash of adding the author and then filtering him out might be confusing/dangerous.

How about we only add him if needed:

```
if (countAuthor) {
  ruleApprovals.push(this.prApi.getAuthor());
}
```

Comment (Bullrich) [reply src/runner.ts:239]: Sounds good. Could you look at the new commit to see if it aligns with what you thought?

File: src/runner.ts [hunk 2]
```diff
@@ -225,13 +225,17 @@ export class ActionRunner {
...
+        requiredApprovals: reviewers.min_approvals,
+        countAuthor: reviewers.countAuthor ?? false,
+      });
     }
 
     // We count how many reviews are needed in total
     const requiredAmountOfReviews = rule.reviewers.map((r) => r.min_approvals).reduce((a, b) => a + b, 0);
     // We get the list of users that approved the PR
-    const approvals = await this.prApi.listApprovedReviewsAuthors();
+    const approvals = await this.prApi.listApprovedReviewsAuthors(false);
```
Comment (rzadp) [inline src/runner.ts:238]: Actually why don't we do this here?

```suggestion
    const approvals = await this.prApi.listApprovedReviewsAuthors(rule.countAuthor ?? false);
```

Comment (Bullrich) [reply src/runner.ts:238]: It's not possible I'm afraid, as rule doesn't have the `countAuthor` property.

It is nested as `rule.reviewers[].countAuthor` so I need to extract it during that iteration.

Comment (rzadp) [reply src/runner.ts:238]: Ahh right, I get it now.

Comment (Bullrich) [reply src/runner.ts]: `rule.reviewers[].countAuthor` has been moved to `rule.countAuthor` so now it is possible 😄

**Review by rzadp — APPROVED** [review]

**Pull request merged**